### PR TITLE
Add MultiNode TradeFed tests for --shards and with session retry

### PR DESCRIPTION
**Pull request by kateyy**
Continuation of changes previously discussed on Gerrit: [tradefed_shards_with_retry](https://review.linaro.org/#/q/topic:tradefed_shards_with_retry+(status:open+OR+status:merged))

### CODE REVIEW

**Comment by kateyy** [comment]
What is the policy to fix W503 and W504; do you prefer to violate a maximum line length? Why not using black formatting as for LAVA itself? (Which is not fully compatible with pycodestyle).

File: automated/android/multinode/tradefed/example-job-template-cts.yaml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+- test:
+    namespace: tlxc
+    role:
+    - master
+    - worker
+    timeout:
+      minutes: 20
+    definitions:
+    - repository: https://review.linaro.org/qa/test-definitions
```
Comment (chase-qi) [inline automated/android/multinode/tradefed/example-job-template-cts.yaml]: Can you please replace repository url with https://github.com/Linaro/test-definitions ?

**Comment by chase-qi** [comment]
Sanity check is line length flexible already https://github.com/Linaro/test-definitions/blob/master/sanity-check.sh#L7 . Introducing black formatting needs more investigation, I don't know if LAVA using it. @mwasilew ideas/comments? I personally hope we are more transparent, less solid black.

And please ignore the failure with job example. We need some magic to distinguish between test job and test definition yaml files. Tracking with https://github.com/Linaro/test-definitions/issues/17

**Comment by mwasilew** [comment]
@chase-qi lava uses 'black' formatting now. In our case I see not much benefit to be honest but I have nothing against moving to this code style.

**Comment by chase-qi** [comment]
Again, thanks for contributing. I am merging the PR.

**Pull request merged**